x-pack/filebeat/input/etw/input.go

```diff
@@ -0,0 +1,148 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+
+package etw_input
+
+import (
+	"fmt"
+	"sync"
+	"syscall"
+	"time"
+
+	input "github.com/elastic/beats/v7/filebeat/input/v2"
+	stateless "github.com/elastic/beats/v7/filebeat/input/v2/input-stateless"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	"github.com/elastic/beats/v7/x-pack/libbeat/reader/etw"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+const (
+	inputName = "etw"
+)
+
+type etw_input struct {
+	log        *logp.Logger
+	config     config
+	etwSession etw.Session
+}
+
+func Plugin() input.Plugin {
+	return input.Plugin{
+		Name:      inputName,
+		Stability: feature.Beta,
+		Info:      "Collect ETW logs.",
+		Manager:   stateless.NewInputManager(configure),
+	}
+}
+
+func configure(cfg *conf.C) (stateless.Input, error) {
+	conf := defaultConfig()
+	if err := cfg.Unpack(&conf); err != nil {
+		return nil, err
+	}
+
+	return newETW(conf)
+}
+
+func newETW(config config) (*etw_input, error) {
+	if err := config.validate(); err != nil {
+		return nil, err
+	}
+
+	return &etw_input{
+		config: config,
+	}, nil
+}
+
+func (e *etw_input) Name() string { return inputName }
+
+func (e *etw_input) Test(_ input.TestContext) error {
+	// ToDo
+	return nil
+}
+
+func (e *etw_input) Run(ctx input.Context, publisher stateless.Publisher) error {
+	var err error
+	e.etwSession, err = etw.NewSession(convertConfig(e.config))
+	if err != nil {
+		return fmt.Errorf("error when inicializing '%s' session: %v", e.etwSession.Name, err)
```

Comment:
etwSession.Name always contain a valid content. It is the first step in `etw.NewSession()` and cannot fail because it is filled from some required config parameters, in case these parameters are missing the config validation would fail.